src/plugins/accounting.cpp

```diff
@@ -103,10 +103,71 @@ json_t* Association::to_json () const
         }
     }
 
+    json_t *hj_json = json_object ();
+    if (!hj_json) {
+        json_decref (user_queues);
+        json_decref (user_projects);
+        json_decref (queue_usage_json);
+        return nullptr;
+    }
+
+    for (const auto &entry : hj) {
+        const Job &job = entry.second;
+        json_t *job_json = json_pack ("{s:i, s:i, s:s, s:o}",
+                                      "nnodes", job.nnodes,
+                                      "ncores", job.ncores,
+                                      "queue", job.queue.c_str (),
+                                      "deps", json_array ());
+
+        if (!job_json) {
+            json_decref (user_queues);
+            json_decref (user_projects);
+            json_decref (queue_usage_json);
+            json_decref (hj_json);
+            return nullptr;
+        }
+
+        json_t *deps_array = json_array ();
+        if (!deps_array) {
+            json_decref (user_queues);
+            json_decref (user_projects);
+            json_decref (queue_usage_json);
+            json_decref (hj_json);
+            json_decref (job_json);
+            return nullptr;
+        }
+
+        for (const auto &dep : job.deps) {
+            json_t *dep_str = json_string (dep.c_str ());
+            if (!dep_str || json_array_append_new (deps_array, dep_str) < 0) {
+                json_decref (user_queues);
+                json_decref (user_projects);
+                json_decref (queue_usage_json);
+                json_decref (hj_json);
+                json_decref (deps_array);
+                json_decref (job_json);
+                return nullptr;
+            }
+        }
+
+        json_object_set_new (job_json, "deps", deps_array);
+
+        // use job id (flux_jobid_t) as string key
+        char keybuf[32];
+        snprintf (keybuf, sizeof(keybuf), "%ld", entry.first);
+        if (json_object_set_new (hj_json, keybuf, job_json) < 0) {
+            json_decref (user_queues);
+            json_decref (user_projects);
+            json_decref (queue_usage_json);
+            json_decref (hj_json);
```

Comment:
Should decref `job_json` here too? Because if this function returns -1, I think you can assume the reference wasn't stolen.